### PR TITLE
[Validator] allow passing a validator to Validation::createCallable()

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 5.1
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

As spotted by @stof in https://github.com/symfony/symfony/pull/31466#issuecomment-630054227

### CODE REVIEW

**Comment by xabbuh** [comment]
I think we should also throw a `TypeError` if one calls the method with `null` as the first argument followed by some constraints.

**Comment by nicolas-grekas** [comment]
If we don't allow null, it means the signature cannot be expressed using regular PHP types and we must trick it using func_numb_args(). I think we'd better avoid such things.

**Review by xabbuh — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Component/Validator/Validation.php [hunk 1]
```diff
@@ -23,10 +23,21 @@ final class Validation
...
+    public static function createCallable($constraintOrValidator = null, Constraint ...$constraints): callable
     {
-        $validator = self::createValidator();
+        $validator = $constraintOrValidator;
+
+        if ($constraintOrValidator instanceof Constraint) {
+            $constraints = \func_get_args();
+            $validator = null;
+        } elseif (null !== $constraintOrValidator && !$constraintOrValidator instanceof ValidatorInterface) {
+            throw new \TypeError(sprintf('Argument 1 passed to "%s()" must be a "%s" or a "%s" object, "%s" given.', __METHOD__, Constraint::class, ValidatorInterface::class, get_debug_type($constraintOrValidator)));
```
Comment (stof) [inline src/Symfony/Component/Validator/Validation.php:37]: The error message should say `or "null"` too.

**Review by stof — APPROVED** [review]

File: src/Symfony/Component/Validator/Validation.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Validator/Validation.php:37]: no need to advertise this case IMHO

**Pull request merged**